python/lsst/ip/isr/shutterMotion.py

```diff
@@ -0,0 +1,576 @@
+        # start and final position.  Equation (5.2).
+        V1 = t1**2 * (j0 - j1)/2. - t1*A1
+        S1 = t1**3 * (j0 - j1)/6. - t1**2 * A1/2. - t1*V1
+        Smid = 0.5*(self.metadata['startPosition'] + self.metadata['endPosition'])
+
+        def pos(t):
+            return j1*(t**3)/6. + A1*(t**2)/2. + V1*t + S1 - Smid
+
+        try:
+            tm_position = newton(pos, tm_accel)
+        except Exception as e:
+            self.log.warn(f"Midpoint calculation (from position) failed to converge: {e}")
+            tm_position = nan
+
+        # Restore t0 so these can be compared to raw timestamps.
+        return tm_accel + t0, tm_position + t0
+
+    @classmethod
+    def fromDict(cls, dictionary):
+        """Construct a ShutterMotionProfile from a dictionary of properties.
+
+        Parameters
+        ----------
+        dictionary : `dict`
+            Dictionary of properties.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.ShutterMotionProfile
+            Constructed calibration.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if the supplied dictionary is for a different
+            calibration type.
+        """
+        calib = cls()
+
+        if calib._OBSTYPE != dictionary['fileType']:
+            raise RuntimeError(f"Incorrect calibration supplied.  Expected {calib._OBSTYPE}, "
+                               f"found {dictionary['OBSTYPE']}")
+        motionProfile = dictionary.pop('motionProfile')
+
+        encodeSamples = motionProfile.pop("encodeSamples")
+        hallTransitions = motionProfile.pop("hallTransitions")
+        fitResults = motionProfile.pop("fitResults")
+
+        if 'metadata' in dictionary:
+            metadata = dictionary.pop('metadata')
+            for key, value in metadata.items():
+                dictionary[key] = value
+        calib.setMetadata(dictionary)
+
+        formatVersion = calib.metadata['version']
+
+        startTime = motionProfile.pop("startTime")
+        if formatVersion == 1.0:
+            motionProfile["startTime_tai"] = startTime["tai"]
+            motionProfile["startTime_mjd"] = startTime["mjd"]
+        else:
+            motionProfile["startTime_tai"] = startTime["tai"]["isot"]
+            motionProfile["startTime_mjd"] = startTime["tai"]["mjd"]
+
+        calib.readEncodeSamples(encodeSamples, formatVersion)
+        calib.readHallTransitions(hallTransitions, formatVersion)
+        calib.readFitResults(fitResults)
+
+        calib.updateMetadata(**motionProfile)
+        return calib
+
+    def toDict(self):
+        """Return a dictionary containing the calibration properties.
+
+        The dictionary should be able to be round-tripped through
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+        formatVersion = self.metadata['version']
+
+        if formatVersion == 1.0:
```

Comment:
Can you put a bit more about what the different format versions are about?

Reply:
```            # Original format.```
and
```            # Update to clarify all times are in the TAI system.```
Possibly not the most detailed, but this was really just a switch to ensure that it's clear that both the ISO and MJD are TAI based.